pandas/io/common.py

```diff
@@ -126,9 +127,17 @@ def _validate_header_arg(header) -> None:
         )
 
 
-def _stringify_path(
-    filepath_or_buffer: FilePathOrBuffer[AnyStr]
-) -> FilePathOrBuffer[AnyStr]:
+@overload
+def _stringify_path(filepath_or_buffer: Union[str, Path]) -> str:
+    ...
```

Comment:
Does ellipses vs pass matter here?  AFAIK could be “overload” magic

Reply:
I think this is just a typing convention:

https://mypy.readthedocs.io/en/latest/stubs.html#stub-file-syntax

^ noted for stub files but I think OK here as well

Reply:
see https://docs.python.org/3/library/typing.html#typing.overload for example.

Reply:
from https://mypy.readthedocs.io/en/latest/more_types.html#runtime-behavior...

"While we can leave the variant body empty using the pass keyword, the more common convention is to instead use the ellipsis (...) literal."

@jbrockmendel mark as resolved?